### PR TITLE
	Complete SQL parameter substitution.

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

**Comment by tschottdorf** [comment]
looks like you take out `walkTableExpr` in the first commit (whose message I can't parse, btw) and put it back in in the second. That doesn't seem useful.

**Comment by tschottdorf** [comment]
sorry, read that backwards. That's ok.

**Comment by tschottdorf** [comment]
LGTM

File: sql/parser/walk.go [hunk 1]
```diff
@@ -183,13 +189,33 @@ func WalkStmt(v Visitor, stmt Statement) {
...
-			for i, tuple := range rows {
-				rows[i] = WalkExpr(v, tuple).(Tuple)
+		for i, expr := range stmt.GroupBy {
+			stmt.GroupBy[i] = WalkExpr(v, expr)
+		}
+		if stmt.Having != nil {
+			stmt.Having.Expr = WalkExpr(v, stmt.Having.Expr)
+		}
+		for i, expr := range stmt.OrderBy {
+			stmt.OrderBy[i].Expr = WalkExpr(v, expr.Expr)
```
Comment (bdarnell) [inline sql/parser/walk.go]: Many databases do not allow substitution in `ORDER BY` or other clauses referencing columns (on the other hand, users want parameterized `ORDER BY`, and doing this outside the database's placeholder system is a common source of SQL injection). The idea is that placeholders are only for values; the skeleton of the query should be sufficient for constructing a query plan without filling in arguments (so this plan can be constructed once and reused for many sets of values).

Comment (vivekmenezes) [inline sql/parser/walk.go]: What we've implemented is what is supported by the postgres grammer.

**Comment by vivekmenezes** [comment]
https://github.com/cockroachdb/cockroach/issues/1616

**Pull request merged**